vendor/github.com/intel/govmm/qemu/qemu.go

```diff
@@ -385,6 +385,9 @@ func (fsdev FSDevice) QemuParams(config *Config) []string {
 		deviceParams = append(deviceParams, fmt.Sprintf(",romfile=%s", fsdev.ROMFile))
 	}
 	if fsdev.Transport.isVirtioCCW(config) {
+		if config.Knobs.IOMMUPlatform {
```

Comment:
@huoqifeng Please submit this change to https://github.com/intel/govmm

Reply:
Thanks for the comments, I'm current busy on other stuff, will check it and revise later...

Reply:
Once the change lands in govmm, you then need to "re-vendor" it into this repo by creating a separate commit to the files changed below `vendor/`. That commit should list the changes in the upstream project. We've documented how to do this here:

https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs

It might sound complex but actually isn't :smile: Just ask us if unsure though.

Reply:
PR https://github.com/intel/govmm/pull/132 created in repo govmm.

Reply:
Created separate PR https://github.com/kata-containers/runtime/pull/2871 to revendor govmm, but it fails the travis check, the details is described in the new PR. @jodh-intel any suggestion?

Reply:
Hi @huoqifeng - There are a few problems here, but don't worry they are all easy to resolve :smile:

First, if your PR needs to revendor one of the vendored packages like govmm, you can simply create a single commit as part of your main PR to do the revendor. By convention, the [subsystem](https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#subsystem) (which is the first token up to the colon on the first line of the commit) is "vendor".

As the link above shows, the subsystem *must* be a single word so that is one of reasons Travis is failing as you have specified two ("cf0f05d" and "qemu"):

```
cf0f05d qemu: add iommu_platform knob for qemuParams
```

The final issue is that the commit on #2871 doesn't have a body explaining the change in more details. We have a standard format for revendor commits: to list all the commits in that upstream project in the commit body (see `old_commit=...; new_commit=...` in this section: https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#re-vendor-prs

In summary, you need to do the following:

1) Add the revendor commit to this PR.
2) Make that revendor commit look something like this:

    ```
    vendor: Update govmm for s390x iommu_platform annoations

    Updated govmm to support iommu_platform annotations for virtio devices. govmm changes:

       cf0f05d qemu: add iommu_platform knob for qemuParams

    Signed-off-by: ...
    ```
3) Close #2871 since the change is now part of this PR.

----

See https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format for further details.

Reply:
@jodh-intel thanks, it helps!

Reply:
I suffered very slow `dep ensure` command, it's no help even I deleted Gopkg.lock file and the obsolete govmm vendor code.